x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -266,7 +267,7 @@ public void authorizeIndexAction(RequestInfo requestInfo, AuthorizationInfo auth
                     // information such as the index and the incoming address of the request
                     listener.onResponse(new IndexAuthorizationResult(true, IndicesAccessControl.ALLOW_NO_INDICES));
                 }
-            } else if (isAsyncSearchRelatedAction(action)) {
+            } else if (isAsyncSearchRelatedAction(action) || isAsyncEqlSearchRelatedAction(action)) {
```

Comment:
Maybe add a `isAsyncAction` to encapsulate the condition of both search and eql searches there? Any future updates would not touch `authorizeIndexAction`.